mina-daemon/templates/deployment.yaml

```diff
@@ -392,18 +416,74 @@ spec:
             - name: {{ .Release.Name }}-fixed-mina-keys
               mountPath: /root/mina-keys
             {{- end }}
-            {{- if .Values.deployment.uploadBlocksToGCloud }}
+            {{- if or .Values.deployment.storeBlocks.gcp.enabled .Values.deployment.storeBlocks.aws.enabled }}
             - name: {{ .Release.Name }}-gcloud-keyfile
               mountPath: "/gcloud/"
             {{- end }}
             {{- if .Values.deployment.genesisLedgerURL }}
             - name: {{ .Release.Name }}-genesis-ledger-dir
               mountPath: /root/.mina-config
             {{- end }}
-            {{- if .Values.deployment.storeBlocks.storageClass }}
+            {{- if .Values.deployment.storeBlocks.enabled }}
             - name: {{ .Release.Name }}-precomputed-blocks
               mountPath: {{ .Values.deployment.storeBlocks.directory }}
             {{- end }}
+            {{- if .Values.deployment.storeBlocks.aws.enabled }}
+            - name: {{ .Release.Name }}-scripts
+              mountPath: /usr/bin/gsutil
+              subPath: gsutil-impersonator.sh
```

Comment:
I like that you kept a clear separation of what is a script and what it impersonates!